src/parsers/vcCSV.cpp

```diff
@@ -0,0 +1,290 @@
+
+    // factor in any remainder from last chunk
+    int64_t actualChunkLength = readChunkLength + bytesRemaining;
+
+    // avoid mangled read of last data in the chunk
+    if (actualChunkLength < fileLength)
+    {
+      // nul terminate on the last newline, and factor any remaining data into the start of the next chunk
+      size_t lastEntryNewline = 0;
+      udStrrchr(chunkMemory, "\n", &lastEntryNewline);
+      chunkMemory[lastEntryNewline] = '\0';
+
+      bytesRemaining = actualChunkLength - (int64_t)(lastEntryNewline + 1); // skip the newline character
+    }
+
+    for (char *pLine = strtok(chunkMemory, "\r\n"); pLine; pLine = strtok(nullptr, "\r\n"))
+    {
+      if (vcCSV_ShouldSkipLine(pLine))
+        continue;
+
+      // skip entries if specified
+      if (pCSV->importSettings.skipEntries > pCSV->skippedEntries)
+      {
+        ++pCSV->skippedEntries;
+        continue;
+      } 
+
+      if (*pLine)
+      {
+        pCSV->entryCount++;
+
+        if (pDelimeter != nullptr)
+        {
+          char *pTokenArray[MAX_COLUMN_COUNT] = {};
+          int count = udStrTokenSplit(pLine, pDelimeter, pTokenArray, MAX_COLUMN_COUNT);
+
+          pCSV->elementCount += count;
+          for (int c = 0; c < count; ++c)
+          {
+            int64_t dataLen = (int64_t)udStrlen(pTokenArray[c]) + 1; // include null terminator
+            vcCSV_AppendData(pCSV, pTokenArray[c], dataLen);
+          }
+        }
+        else
+        {
+          // assumed its fixed spacing
+          int count = (int)udStrlen(pLine) / pCSV->importSettings.fixedSizeDelimeterSpacing;
+          pCSV->elementCount += count;
+          for (int c = 0; c < count; ++c)
+          {
+            int dataLen = pCSV->importSettings.fixedSizeDelimeterSpacing;
+            int offset = (c * pCSV->importSettings.fixedSizeDelimeterSpacing);
+            vcCSV_AppendData(pCSV, pLine + offset, dataLen);
+            vcCSV_AppendData(pCSV, "\0", 1); // append null terminator between data
+          }
+        }
+      }
+    }
+
+    totalBytesRead += actualChunkLength - bytesRemaining;
+    if (totalBytesRead >= readAmountBytes)
+      break;
+
+    if (bytesRemaining > 0)
+    {
+      // move
+      size_t bytesRead = actualChunkLength - bytesRemaining;
+      memmove(chunkMemory, chunkMemory + bytesRead, bytesRemaining);
+      memset(chunkMemory + bytesRemaining, 0, actualChunkLength - bytesRemaining);
+    }
+  }
+
+  pCSV->totalBytesRead += totalBytesRead;
+
+  if (!pCSV->cancel)
+    result = udR_Success;
+  else
+    result = udR_Cancelled;
+
+  if (result == udR_Success && (int64_t)pCSV->totalBytesRead == fileLength)
+    pCSV->completeRead = true;
+
+epilogue:
+
+  if (pCSV)
+    pCSV->readResult = result;
+
+  udFile_Close(&pFile);
+
+  udFree(pDelimeter);
+  return result;
+}
+
+udResult vcCSV_ReadHeader(vcCSV *pCSV)
+{
+  udResult result;
+
+  while (!pCSV->completeRead)
+  {
+    const size_t minimumHeaderReadSize = CAPACITY_CHUNK_SIZE; // 8kb
```

Comment:
You should put this comment with the `CAPACITY_CHUNK_SIZE`, also I think it's actually 16kb?